python/cudf/cudf/_lib/pylibcudf/column.pyx

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from cython.operator cimport dereference
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+from libcpp.vector cimport vector
+
+from rmm._lib.device_buffer cimport DeviceBuffer
+
+from cudf._lib.cpp.column.column cimport column, column_contents
+from cudf._lib.cpp.types cimport bitmask_type, offset_type, size_type
+
+from .gpumemoryview cimport gpumemoryview
+from .types cimport DataType
+from .utils cimport int_to_bitmask_ptr, int_to_void_ptr
+
+
+cdef class Column:
+    """A container of nullable device data as a column of elements."""
+    def __init__(
+        self, DataType data_type not None, size_type size, gpumemoryview data,
+        gpumemoryview mask, size_type null_count, offset_type offset,
```

Comment:
I'd like to avoid putting any attributes into a `gpumemoryview` that don't exist in a Python [`memoryview`](https://docs.python.org/3/library/stdtypes.html#memoryview) so that if we do eventually try to align/merge them we won't have too much difficulty.

Regarding the offset applying to the mask and the data, yes it applies to both. That's the same behavior as in libcudf, so I don't know if there's any additional documentation to do here, but happy to add if you have thoughts.

Reply:
> So I guess one way to do it would be to construct the memoryview from ptr + offset, rather than ptr. Then one would remove the need (I think) for columns to have an offset at all

That would only work for fixed width data types, no?

Reply:
Right. I think you could make it work for nested types but the math would become nontrivial.